### DISCUSSION TITLE
route for /.well-known fails to match

### DISCUSSION THREAD

**Discussion by thinkmorestupidless** [playframework/playframework#11783] [Q&A]
I'm attempting to set up universal links for iOS and need to serve a file from `/.well-known/apple-app-site-association` - but this line in my route file

```
GET     /.well-known/*file         controllers.Assets.versioned(path="/public/.well-known", file: Asset)
```

doesn't seem to want to match

if i remove the period (so the path is `/well-known/*file`) then it matches...

is this me missing something obvious?

**Comment by pepite** [comment]
I think you need to escape the . (dot) as I seem to remember it is a sort
of regular expression (ie *file for example).

On Mon, 1 May 2023 at 18:10, Trevor Burton-McCreadie <
***@***.***> wrote:

> I'm attempting to set up universal links for iOS and need to serve a file
> from /.well-known/apple-app-site-association - but this line in my route
> file
>
> GET     /.well-known/*file         controllers.Assets.versioned(path="/public/.well-known", file: Asset)
>
> doesn't seem to want to match
>
> if i remove the period (so the path is /well-known/*file) then it
> matches...
>
> is this me missing something obvious?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/playframework/discussions/11783>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AABMF7LVSMKHI4UKESGZNADXD7N6HANCNFSM6AAAAAAXR5HLFM>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
Nicolas Leroux

**Reply by thinkmorestupidless** [reply]
makes sense... but i'm confused about _how_ i'd escape that - simply putting `/\.well-known/*file` results in `invalid escape character`

and i can't create a dummy path segment ( with `/$period<\.>well-known`) because that will make the router expect an argument named `period` in the handler...

**Reply by slisaasquatch** [reply]
I don't have the answer to the original question, but I don't think the dot needs to be escaped, because I have a route myself defined like this and it works.
```
GET		/.well-known/jwks.json		com.example.MyController.jwks()
```

**Comment by cyrilfr** [comment]
I faced the same issue and I just removed the dot in the name of the folder:

    GET     /.well-known/*file                          controllers.Assets.at(path="/public/well-known", file: Asset)
